### PR TITLE
Shadowcasting field of view + enhanced fog of war.

### PR DESCRIPTION
**Pull request by ipochto**
- Added enhanced fog of war preferences. Enhanced type is enabled by default.
- Added field of view settings. Standard radial is enabled by default and will be reloaded after every game. It allows to change FOV type for certain maps (need to enable required FOV type in the map lua-file).
- Auto enabled shadow casting field of view type for maps with dungeon tileset.

Corresponded PR:
https://github.com/Wargus/stratagus/pull/330

### CODE REVIEW

**Comment by timfel** [comment]
This looks really good, especially in the dungeon levels! The only concern I have is the high CPU load. I feel this should be a preference that is off by default, just like the shaders, so we don't immediately loose people who are e.g. on a Raspberry Pi :)

**Comment by ipochto** [comment]
Do you mean noted high CPU load on the PC or Raspberry Pi? Because I've made a lot of profiling and cpu load was reduced radically.  But I tested on a processor with 8 cores... We can dramatically improve performance with gpu-side renderer (not sure if Raspberry supports it). Just now fog generation and bilinear interpolation are on the CPU-side.

Thought to leave the possibility to change this parameter for developers. But if it really puts so much load on the processor, we'll have to give the user this option.
Thought to leave the possibility to change this parameter for developers. But if it really puts so much strain on the processor, we'll have to give the user this option.

**Comment by timfel** [comment]
> Thought to leave the possibility to change this parameter for developers. But if it really puts so much strain on the processor, we'll have to give the user this option.

Yes, I think we need to give the user the option. On my 8 core i7 from 2017 I experienced a few slowdowns in War1gus and the load was consistently at 100% for 4 cores. Definitely will be too slow on any older or weaker hardware :)

**Comment by ipochto** [comment]
It's very strange. I tested it on i7 from 2011 in high resolution (-v 1280x1024). There were no slowdowns for war1gus. For wargus was, but not for war1gus.
And why only 4 cores was used? I don't make any restriction for number of cores. All cores were to be used.
Could you please provide more information (resolution, bilinear enabled etc.)?

**Comment by ipochto** [comment]
I've reworked a bit the whole conception of FOV option. There is a check-box preference now to enable or disable sight blocking for dungeons.

image

File: scripts/stratagus.lua [hunk 1]
```diff
@@ -310,7 +310,7 @@ DefaultPreference("MultiColoredCampaigns", true)
 DefaultPreference("ShowButtonPopups", true)
 DefaultPreference("ShowDamage", true)
 DefaultPreference("ShowOrders", true)
-DefaultPreference("OnlineServer", "network.stratagus.de")
+DefaultPreference("OnlineServer", "pvpgn.pro")
```
Comment (timfel) [inline scripts/stratagus.lua:313]: :+1:

**Comment by timfel** [comment]
one thing, should we switch back to using the network.stratagus.de server here?

**Comment by ipochto** [comment]
It's your server so decision should be yours too )

**Comment by ipochto** [comment]
But imho own server is always better )

**Pull request merged**